### PR TITLE
feat: use Flatlist for new homeview

### PR DESCRIPTION
**Pull request by MounirDhahri**
[ONYX-1327]
### Description

This PR updates the `Homeview` to use `Flatlist` instead of `FlashList`

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- use Flatlist in new home view - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1327]: https://artsyproduct.atlassian.net/browse/ONYX-1327?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/HomeView/HomeView.tsx [hunk 1]
```diff
@@ -147,6 +145,7 @@ export const HomeView: React.FC = () => {
           }
           refreshControl={<RefreshControl refreshing={isRefreshing} onRefresh={handleRefresh} />}
           onEndReachedThreshold={1}
+          windowSize={11}
```
Comment (anandaroop) [inline src/app/Scenes/HomeView/HomeView.tsx:148]: **question (non-blocking):** How do we choose this value?

**Review by anandaroop — APPROVED** [review]
Thanks for the quick fix 🙏🏽 

Should we link this to https://artsyproduct.atlassian.net/browse/ONYX-1327?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 70e94690aed4aaff9f625a3c6c62d7b2b9ef2df7
</p>

File: src/app/Scenes/HomeView/HomeView.tsx (hunk 1 above)
Comment (dblandin) [reply src/app/Scenes/HomeView/HomeView.tsx:148]: I have the same question. Found some official documentation [here](https://reactnative.dev/docs/optimizing-flatlist-configuration#windowsize) and we have an unspecific warning about using windowSize [in our best practices doc](https://github.com/artsy/eigen/blob/main/docs/best_practices.md#use-windowsize-with-caution) but without more detail, I'm not really sure what the guidance is.

Is the default windowSize of 21 insufficient for our purpose here?

**Review by dblandin — DISMISSED** [review]

File: src/app/Scenes/HomeView/HomeView.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/HomeView.tsx:148]: Usually you would want to how enough items for about 2/3 screens - for us that's around 5/6 sections from above and 5/6 sections from below. Therefore the window size I went with as 11. I will revert this to be 21 as by default and we can improve it at a later stage

Comment (dblandin) [reply src/app/Scenes/HomeView/HomeView.tsx]: Yeah, maybe we just stick with the default for now, especially as sections can take up different heights today, and we can make more targeted optimizations later given some instrumentation.

**Pull request merged**